doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1462 @@
+.. ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2023, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _ARKODE.Usage.SPRKStep.UserCallable:
+
+SPRKStep User-callable functions
+==================================
+
+This section describes the functions that are called by the
+user to setup and then solve an IVP using the SPRKStep time-stepping
+module. Some of these are required; however, starting with
+:numref:`ARKODE.Usage.SPRKStep.OptionalInputs`, the functions listed involve
+optional inputs/outputs or restarting, and those paragraphs may be
+skipped for a casual use of ARKODE's SPRKStep module. In any case,
+refer to the preceding section, :numref:`ARKODE.Usage.SPRKStep.Skeleton`,
+for the correct order of these calls.
+
+On an error, each user-callable function returns a negative value  (or
+``NULL`` if the function returns a pointer) and sends an error message
+to the error handler routine, which prints the message to ``stderr``
+by default. However, the user can set a file as error output or can
+provide her own error handler function (see
+:numref:`ARKODE.Usage.SPRKStep.OptionalInputs` for details).
+
+
+
+.. _ARKODE.Usage.SPRKStep.Initialization:
+
+SPRKStep initialization and deallocation functions
+------------------------------------------------------
+
+
+.. c:function:: void* SPRKStepCreate(ARKRhsFn f1, ARKRhsFn f2, realtype t0,\
+                                     N_Vector y0, SUNContext sunctx)
+
+   This function allocates and initializes memory for a problem to
+   be solved using the SPRKStep time-stepping module in ARKODE.
+
+   :param f1: the name of the C function (of type :c:func:`ARKRhsFn()`) defining :math:`f_1(q,t) = \frac{\partial V(q,t)}{\partial q}`
+   :param f2:  the name of the C function (of type :c:func:`ARKRhsFn()`) defining :math:`f_2(p) = \frac{\partial T(p)}{\partial p}`
```

Comment:
Similarly, `f1` and `f2` should fill in disjoint sets of variables in `ydot` if I understand correctly. And `ydot` has length equal to the full dimension of the ODE? This documentation implies that `f1` and `f2` return subvectors for the `p` and `q` components only.

Reply:
I updated the documentation to clarify that the vector returned is the full vector including `p` and `q`. If we wanted to make it so that it only the `p` and `q` components were sent to the Rhs functions, we would need to require the use of the ManyVector. 

There are no internal tests to ensure that f2 is independent of `t`.  What we send in is the `t` value sent into `f1` which is the `tn + ci * h`.

Reply:
`f1` is passed `ark_mem->tn + chati*ark_mem->h` and `f2` is passed `ark_mem->tn + ci*ark_mem->h`. Is it usually the case that `chati = ci`?

The notation for comments and documentation is inconsistent about having `f2` as `f(t, p)` or `f(p)`. I think I comment on most of these to add the `t` but maybe it better to leave it off. Thoughts?

Reply:
I think we should add the `t` as it *should* work. `chati /= ci` in general. I see in the math section I flipped (typo) the use of \hat and no hat between the regular and incremental form.

Reply:
Ill note that the original comment here was from before I made it possible to have a time-dependent f1 and f2.